### PR TITLE
Added accountDB storage and test accountDB files

### PR DESCRIPTION
**Pull request by stasBigunenko**
Create new accountDB in memory storage

### CODE REVIEW

File: pkg/storage/accountsDB/accountInterface.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	"github.com/stasBigunenko/monorepa/model"
+)
+
+type AccountsService interface {
+	GetAccount(context.Context, uuid.UUID) (model.Account, error)
+	GetUserAccounts(context.Context, uuid.UUID) ([]model.Account, error)
+	GetAllUsers(context.Context) ([]model.Account, error)
+	CreateAccount(context.Context, uuid.UUID) (model.Account, error)
+	UpdateAccount(context.Context, model.Account) (model.Account, error)
+	DeleteAccount(context.Context, uuid.UUID) error
```
Comment (DrIhor) [inline pkg/storage/accountsDB/accountInterface.go:15]: Names can be same as at `User(Get, Update, etc)`

File: pkg/storage/accountsDB/accountDB.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	as.mu.Lock()
+	accounts := []model.Account{}
+
+	for _, account := range as.MapAccount {
+		if userID == account.UserID {
+			accounts = append(accounts, account)
+		}
+	}
+
+	as.mu.Unlock()
```
Comment (DrIhor) [inline pkg/storage/accountsDB/accountDB.go:43]: Can use mutex unlock with `defer` after lock

File: pkg/storage/accountsDB/accountDB.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	as.MapAccount = make(map[uuid.UUID]model.Account)
+	return &as
+}
+
+func (as *AccountStorage) GetAccount(_ context.Context, id uuid.UUID) (model.Account, error) {
+	as.mu.Lock()
+	val, ok := as.MapAccount[id]
+	as.mu.Unlock()
+	if !ok {
+		return model.Account{}, fmt.Errorf("account with id %d not found", id)
```
Comment (DrIhor) [inline pkg/storage/accountsDB/accountDB.go:27]: Return error or empty value without error??? Other DB can return many type of errors and handle empty data can be difficult.

Comment (alex-a-renoire) [reply pkg/storage/accountsDB/accountDB.go:27]: Do you mean to add custom error type?

File: pkg/storage/accountsDB/accountDB_test.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+	t.Run("GetUserAccounts", func(t *testing.T) {
+		accountsSet := []model.Account{
+			account2,
+			account3,
+		}
+
+		val, _ := acc.GetUserAccounts(context.Background(), userID2)
+
+		require.Equal(t, accountsSet, val, "should be slice of model.Account with the same userID")
+	})
```
Comment (DrIhor) [inline pkg/storage/accountsDB/accountDB_test.go:57]: Can use table tests(array of tests and results)

File: pkg/storage/accountsDB/accountDB.go (hunk 2 above)
Comment (alex-a-renoire) [reply pkg/storage/accountsDB/accountDB.go:43]: PS. Change all occurances

File: pkg/storage/accountsDB/accountDB.go (hunk 3 above)
Comment (alex-a-renoire) [reply pkg/storage/accountsDB/accountDB.go:27]: PS. Change all occurances

File: pkg/storage/accountsDB/accountInterface.go (hunk 1 above)
Comment (alex-a-renoire) [reply pkg/storage/accountsDB/accountInterface.go:15]: I think it's better to write full function description. It's  more readable.

File: pkg/storage/accountsDB/accountInterface.go [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import (
+	"context"
+	"github.com/google/uuid"
+	"github.com/stasBigunenko/monorepa/model"
+)
+
+type AccountsService interface {
+	GetAccount(context.Context, uuid.UUID) (model.Account, error)
+	GetUserAccounts(context.Context, uuid.UUID) ([]model.Account, error)
+	GetAllUsers(context.Context) ([]model.Account, error)
```
Comment (DrIhor) [inline pkg/storage/accountsDB/accountInterface.go:12]: Do not use method `GetAllUsers` here.

File: pkg/storage/accountsDB/accountInterface.go (hunk 1 above)
Comment (DrIhor) [reply pkg/storage/accountsDB/accountInterface.go:15]: So it must be with `User` service too?)

File: pkg/storage/accountsDB/accountDB.go (hunk 3 above)
Comment (DrIhor) [reply pkg/storage/accountsDB/accountDB.go:27]: Custom error type or when `function() return nil, nil(value, error)` - can`t find.

**Comment by romanilchyshyn** [comment]
[Same](https://github.com/stasBigunenko/monorepa/pull/76#issuecomment-922742584)

First we need a separate pr with generalized storage. After that update these special storages to use it.

**Pull request closed without merging**